DemoApp/Shared/StreamChatWrapper.swift

```diff
@@ -90,18 +90,14 @@ extension StreamChatWrapper {
         // Reset number of refresh tokens
         numberOfRefreshTokens = 0
 
-        // We connect from a background thread to make sure it works without issues/crashes.
-        // This is for testing purposes only. As a customer you can connect directly without dispatching to any queue.
-        DispatchQueue.global().async {
-            self.connect(user: user) { [weak self] error in
-                if let error = error {
-                    log.warning(error.localizedDescription)
-                } else {
-                    self?.onRemotePushRegistration?()
-                }
-                DispatchQueue.main.async {
-                    completion(error)
-                }
+        connect(user: user) { [weak self] error in
+            if let error = error {
+                log.warning(error.localizedDescription)
+            } else {
+                self?.onRemotePushRegistration?()
+            }
+            DispatchQueue.main.async {
```

Comment:
we should already be on the main thread, so this is not needed, right?

Reply:
Actually no, connectUser does not dispatch to the main thread in the completion block

Reply:
Interesting that his one does since most of the completion handlers are called from the main thread. Not part of this PR but sounds like we should change this.